pkg/controller/upload_controller.go

```diff
@@ -327,6 +327,27 @@ func (c *uploadController) processUpload(req *pluginv1api.Upload) error {
 		return errors.New(errMsg)
 	}
 
+	// If upload phase is UploadPhaseCleanupFailed, manually call DeleteLocalSnapshot to retry on deleting local snapshot
+	if req.Status.Phase == pluginv1api.UploadPhaseCleanupFailed {
+		err = c.snapMgr.DeleteLocalSnapshot(peID)
+		if err != nil {
+			errMsg := fmt.Sprintf("Failed to clean up local snapshot, %v, will retry. %v", peID.String(), errors.WithStack(err))
```

Comment:
fmt.Sprintf("Failed to clean up local snapshot %s. Error: %v.  Will retry.", peID.String(), errors.WithStack(err))